---
title: "IODS_Final"
author: "Ashwini Kumar"
date: "8 March 2017"
output: html_document
email: "ashwini.kumar@helsinki.fi"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Data
The data is downloaded from the cBioPortal (http://www.cbioportal.org/index.do) for Cancer Genomics provides visualization, analysis and download of large-scale cancer genomics data sets. 
This data was published in 2013: "Genomic and Epigenomic Landscapes of Adult De Novo Acute Myeloid Leukemia (AML)". The study was published in 2013 NEJM 368:2059-2074 [doi:10.1056/NEJMoa1301689]
Data download link: https://tcga-data.nci.nih.gov/docs/publications/laml_2012/

Background: In the paper they have mentioned data from genomes of 200 clinically-annotated cases of de novo acute myeloid leukemia occurring in adults, using whole genome sequencing (50 cases) or whole exome sequencing (150 cases), mRNA and miRNA sequencing, and DNA methylation analysis. In this exercise, I am going to focus only on the gene expression data from 177 patient samples. 

The aml samples have been sequenced using next generation sequencing technology including both RNA-seq and Exome-seq. I am going to use the gene expression data generated by RNA-seq. RNA-seq, uses next-generation sequencing (NGS) to reveal the presence and quantity of RNA in a biological sample at a given moment. It generates millions of reads and then reads are aligned to referecne human genome to quantify each gene expression. Finally, we get normalized RPKM (Reads Per Kilobase of gene per Million), count up the total reads in a sample and divide that number by 1,000,000 – this is our “per million” scaling factor. Higher the RPKM value explains the higher expression of a given gene. 

I will use log transformed rpkm valuse. The log-transformation is widely used in biomedical and psychosocial research to deal with skewed data. The log transformation can decrease the variability of data and make data conform more closely to the normal distribution. My main pursopse here is to reduce the dimensionality and cluster the patients samples based on the simillar gene expression profiles. 

```{r}
rpkm <- read.csv("log2_rpkm_TCGA_aml.csv", row.names = 1)

dim(rpkm)
class(rpkm)

```
### Patient to patient correlation

```{r}
library(pheatmap)
pheatmap(cor(rpkm), show_rownames = F, show_colnames = F)
```
We have expression values of 19964 genes in 177 samples. Heatmaps explains, the expected positive correlation among patients as we don't have data from healthy samples to compare. All patients would positively correlative with each other as they all have same disease. We can see sub-group of patients that explain that some patients are more closer to each other than otheres. It might be explained by their biology of the disease, for example patients in one group might have same gene mutataion or simillar cytogenetics. Our aim here is just to overview the data.



### Gene filtering
```{r}

# calculate and plot the mean and variance to filter the lowely expressed genes
v <- apply(rpkm,1,var)
m <- apply(rpkm,1,mean)
plot(m,v, main = "RPKM variance vs. mean")

# filter the genes those have mean expression below 4 

filter1 <- rpkm[v > 4, ] 
dim(filter1)



```
The idea here is to select the genes those have high variance and high expression in all the samples. Here, I have selected genes  varaince > 4, this filtering resulted in 90 genes.  

The best way of gene expression data clustering would be MDSplot, PCA plot or heatmaps. 

Now let's try these methods one by one

### PCA
```{r, fig.width=10, fig.height = 10}
# PCA on non-standardised data
#pca_rpkm <- prcomp(filter1)
#summary(pca_rpkm)
#biplot(pca_rpkm, choices = 1:2, cex = c(0.8, 1), col = c("grey40", "deeppink2"), sub = "RNA-seq gene expression (177 aml patient samples)")

# PCA on standardised data (scaling)
rpkm_scale <- scale(filter1)
#summary(rpkm_scale)

pca_rpkm_scale <- prcomp(t(rpkm_scale))
biplot(pca_rpkm_scale, choices = 1:2, cex = c(1.2, 1.2), col = c("grey40", "deeppink2"), sub = "Scaled RNA-seq gene expression (177 aml patient samples)")

```
The PCA plot explains the hetrogenity of the samples, it seems there are many factors those are affecting the overall clustering of the samples. Some interesting observations, gene CD34 is known to be marker gene for ALM disease, and HBB (hemoglobin) gene, they are all correlate positively with values of PC1. Some HOX family genes are positively associated with PC2. HOX family genes are well known for there over expression in set of AML samples with a unique cytogenetics. This explains that the samples on the top of plot are having HOX gene over expression and they share same molecular features.  
#### MDSplot
```{r, fig.width=10, fig.height = 10}
library(edgeR)
plotMDS(filter1)
plotMDS(t(filter1))
```
Principal component analysis (PCA) and Multidimensional scaling (MDS) are common techniques that are used to visualize high-dimentional data. MDS plot shows that samples are not clustering in separate groups. Plots shows both sample and gene clustering. 


### heatmap
```{r, fig.width = 15, fig.height = 20}
library(pheatmap)
library(RColorBrewer)
# heatmap of the filtered genes
mean_rpkm <- rowMeans(filter1)
RPKM_heatmap <- filter1 - mean_rpkm

#set min and max value
minval <- -3
maxval <- 3
pairs.breaks <- c(seq(minval, maxval, length.out=257))

pheatmap(RPKM_heatmap, breaks = pairs.breaks, color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(256), height = 80 )
```
Heatmap shows how both patients and genes are clustered. There is not a single pattern, there are so many clusters of patients due to disease heterogeneity. 
### K-means 

```{r, fig.width = 10, fig.height = 10}
set.seed(123)
K <- 9

km = kmeans(t(filter1),centers=K)

sv = svd(t(filter1))
U = sv$u
V = sv$v
D = sv$d
Z = t(filter1)%*%V
plot(Z[,1],Z[,2],col=km$cluster,type="n")
text(Z[,1],Z[,2],colnames(filter1),cex=1,col=km$cluster)
cens = km$centers
points(cens%*%V[,1],cens%*%V[,2],col=1:K,pch=16,cex=3)
```

Simillar results were observed by K-means clustering as well




